### PR TITLE
kafka: release v0.2.0

### PR DESCRIPTION
**Pull request by zmalik**
this PR bumps latest docs to `v0.2` so we can release the version `0.2.0` to Kafka

### CODE REVIEW

File: repository/kafka/docs/v0.2/configuration.md
Comment (kensipe) [inline repository/kafka/docs/v0.2/configuration.md]: I don't understand most of this section...   This doesn't explain what the default is IMO.   It doesn't explain how the `--instance` flag is tied to the default zk.

File: repository/kafka/docs/v0.2/configuration.md
Comment (kensipe) [inline repository/kafka/docs/v0.2/configuration.md]: this example does switch the zookeeper path.. but it also provides the zookeeper uri.   Is that necessary?

File: repository/kafka/docs/v0.2/configuration.md
Comment (kensipe) [inline repository/kafka/docs/v0.2/configuration.md]: this doesn't make sense

File: repository/kafka/docs/v0.2/configuration.md
Comment (kensipe) [reply repository/kafka/docs/v0.2/configuration.md]: perhaps it doesn't make sense because although the topic is zk path... the main topic is kafka.    If the `--instance` flag is for the installation of zk.. that is unclear.   It is also unclear if that is the case how that is tied to the installation of kafka (unless we do the next step in this section... however this is stated as a default)

**Review by kensipe — CHANGES_REQUESTED** [review]
this looks great... I find some of the docs confusing.  I will follow up with an operator test.

File: repository/kafka/docs/v0.2/configuration.md
Comment (zmalik) [reply repository/kafka/docs/v0.2/configuration.md]: please read the above sentence `To override the default zk path`

File: repository/kafka/docs/v0.2/configuration.md
Comment (zmalik) [reply repository/kafka/docs/v0.2/configuration.md]: so you don't understand `--instance` flag?

File: repository/kafka/docs/v0.2/configuration.md
Comment (zmalik) [reply repository/kafka/docs/v0.2/configuration.md]: thats a typo! its fixed now.

File: repository/kafka/docs/v0.2/configuration.md
Comment (zmalik) [reply repository/kafka/docs/v0.2/configuration.md]: `Zookeeper PATH` is a configuration of Kafka. And it has nothing to do with KUDO Zookeeper. It can be KUDO Zookeeper, DC/OS Zookeeper or living in bare metal. As long as the Kafka has access to the Zookeeper URI it will work.

Comment (kensipe) [reply repository/kafka/docs/v0.2/configuration.md]: that's a bizarre question.... I completely understand what the `--instance` flag is...  however your reference to it doesn't make sense in the context of this documentation.   It seems you are trying to explain 3 options... 1) default, 2) path, 3) uri.     The distinction is lost on me as it is documented now.

Comment (kensipe) [reply repository/kafka/docs/v0.2/configuration.md]: And you might add your last comment into the docs...   The fact that you can have a zookeeper any where should be in the docs.

File: repository/kafka/docs/v0.2/configuration.md
Comment (kensipe) [reply repository/kafka/docs/v0.2/configuration.md]: I did read the entire section including the sentence above.   There is no clarity in the doc regarding the value / difference between URI and PATH.   you mention overriding the PATH... but they you override the URI and the PATH.

**Review by kensipe — CHANGES_REQUESTED** [review]

File: repository/kafka/docs/v0.2/configuration.md
Comment (zmalik) [reply repository/kafka/docs/v0.2/configuration.md]: It's not bizarre,  you literally said `If the --instance flag is for the installation of zk.. that is unclear`. Which leads me to think that maybe I had to clarify what the `--instance` flag is for.

> you are trying to explain 3 options... 1) default, 2) path, 3) uri.

I don't see any reference to above URI here. 

> 1) default, 2) path

By default, the Kafka cluster will try to use a zookeeper path same as specified in `--instance` flag. 

Please feel free to open a PR with suggestions that you think will make these docs readability better.

File: repository/kafka/docs/v0.2/configuration.md
Comment (zmalik) [reply repository/kafka/docs/v0.2/configuration.md]: I see this doc format can be confusing for some people. Let me rethink about that. And I will add docs in table format where all configurable parameters are listed. And some examples of how to change a parameter in a KUDO operator.

**Comment by zmalik** [comment]
We agreed on adding the changes introduced in KUDO `0.4.0` to be addressed in this PR. So let's not merge this PR till its unblocked, after `0.4.0` release.

File: repository/kafka/docs/v0.2/configuration.md
Comment (kensipe) [reply repository/kafka/docs/v0.2/configuration.md]: we need to make this readable and understandable before we merge... at this point it is hold up the merge.   Invitations to improve is always welcome but not a reason to allow something to continue that isn't understandable.

File: repository/kafka/docs/v0.2/configuration.md
Comment (samvantran) [inline repository/kafka/docs/v0.2/configuration.md:13]:
```suggestion
            -p ZOOKEEPER_PATH=/<zk_instance_path> \
```

File: repository/kafka/docs/v0.2/configuration.md
Comment (samvantran) [reply repository/kafka/docs/v0.2/configuration.md]: How about we reword? 
```
##### Zookeeper PATH

Users must set the `ZOOKEEPER_PATH` to where they previously installed zookeeper in order to successfully deploy Kafka. 

To get the zookeeper instance name:
''' # this is meant to be triple backticks 
> kubectl get instances

zookeeper-abc123   1h
'''

Then install Kafka with: 
'''
kubectl kudo install kafka --instance=<my_kafka_name> \
  -p ZOOKEEPER_URI=zk-zk-0.zk-hs:2181,zk-zk-1.zk-hs:2181,zk-zk-2.zk-hs:2181 \
  -p ZOOKEEPER_PATH=/zookeeper-abc123
'''
```
I wouldn't even mention the default. Would that even work? That would mean a user would deploy zookeeper under the same name as kafka? Can we make the default behavior assume `/zookeeper` instead? Then it's a little easier to understand and I would either add at the bottom or top of this section:
```
By default, Kafka will assume a user has already installed zookeeper with the name `/zookeeper`. Thus, a user can deploy Kafka more simply with:

'''
kubectl kudo install kafka --instance=my-kafka-cluster \
  -p ZOOKEEPER_URI=zk-zk-0.zk-hs:2181,zk-zk-1.zk-hs:2181,zk-zk-2.zk-hs:2181 
'''
```

File: repository/kafka/docs/v0.2/install.md
Comment (samvantran) [inline repository/kafka/docs/v0.2/install.md:21]: Hm, based on the `ZOOKEEPER_PATH` instructions above, would this work? 
If you install zookeeper under `/zk`, wouldn't installing `kafka` fail because it thinks the `ZOOKEEPER_PATH=/kafka`?

File: repository/kafka/docs/v0.2/versions.md
Comment (samvantran) [inline repository/kafka/docs/v0.2/versions.md:5]: As mentioned in the other PR, I think it's important to mention the compatible KUDO version here as installing the wrong one might cause issues with the Kafka operator.

**Review by samvantran — COMMENTED** [review]
Just a few comments from me but much less as I already did a hefty number on the other PR ;)

File: repository/kafka/docs/v0.2/configuration.md
Comment (samvantran) [reply repository/kafka/docs/v0.2/configuration.md]: Oh wait, doesn't the `ZOOKEEPER_URI` also represent the zk instance name too? So here a user has previously deployed zookeeper as `zk` right? Hence `zk-zk-0, zk-zk-1,...`

Comment (zmalik) [reply repository/kafka/docs/v0.2/configuration.md]: I have reworked this section: 

##### Zookeeper Configuration

KUDO Kafka requires a running ZooKeeper ensemble to perform its own internal accounting and persist cluster topology. 
You can install the KUDO Zookeeper or an alternate Zookeeper to use with KUDO Kafka. 

###### Configuring the Zookeeper connection:

You can configure KUDO Kafka to use Zookeeper using the KUDO Kafka parameter `ZOOKEEPER_URI`
Let's see this with an example:
```
kubectl kudo install kafka --instance=my-kafka-cluster \
  -p ZOOKEEPER_URI=zk-zookeeper-0.zk-hs:2181,zk-zookeeper-1.zk-hs:2181,zk-zookeeper-2.zk-hs:2181 \
```
In the above example KUDO Kafka cluster will connect to the Zookeeper cluster present in `zk-zookeeper-0.zk-hs:2181,zk-zookeeper-1.zk-hs:2181,zk-zookeeper-2.zk-hs:2181`

The Kafka cluster topology is persisted in a Zookeeper node. Every node in Zookeeper's namespace is identified by a path.   

Two Kafka clusters can share the same Zookeeper but sharing the same path can lead to a corrupt state for both Kafka clusters. 
To avoid this to happen, KUDO Kafka persists its cluster topology in zk node with same name as KUDO Kafka instance.

Let's for example see what will the zk path in the above example where we just configured `ZOOKEEPER_URI`:
```
kubectl kudo install kafka --instance=my-kafka-cluster \
  -p ZOOKEEPER_URI=zk-zookeeper-0.zk-hs:2181,zk-zookeeper-1.zk-hs:2181,zk-zookeeper-2.zk-hs:2181 \
```

In the above example KUDO Kafka will connect the Zookeeper present in `zk-zookeeper-0.zk-hs:2181,zk-zookeeper-1.zk-hs:2181,zk-zookeeper-2.zk-hs:2181` and 
will create a Zookeeper node in path `/my-kafka-cluster`.  

###### Using a custom Zookeeper node path:

You can also decide to chose a custom Zookeeper node path, using the KUDO Kafka parameter `ZOOKEEPER_PATH`
Let's see this with an example:
```
kubectl kudo install kafka --instance=my-kafka-name \
  -p ZOOKEEPER_URI=zk-zookeeper-0.zk-hs:2181,zk-zookeeper-1.zk-hs:2181,zk-zookeeper-2.zk-hs:2181 \
  -p ZOOKEEPER_PATH=/custom-path
```
In the above example KUDO Kafka will connect the Zookeeper present in `zk-zookeeper-0.zk-hs:2181,zk-zookeeper-1.zk-hs:2181,zk-zookeeper-2.zk-hs:2181` and 
will create a Zookeeper node in path `/custom-path`.

File: repository/kafka/docs/v0.2/configuration.md
Comment (zmalik) [reply repository/kafka/docs/v0.2/configuration.md:13]: not instance path but zk node for kafka cluster. In DC/OS, for example, its `/dcos-service-kafka` if service name is `kafka`. ` /prod/my-kafka-cluster` becomes a zk node in path `/dcos-service-prod__my-kafka-cluster`.

Same principal applies here. It uses KUDO kafka instance name for zk node path. So having a static default can lead to issues.

File: repository/kafka/docs/v0.2/versions.md
Comment (zmalik) [reply repository/kafka/docs/v0.2/versions.md:5]: something like this: 

### KUDO Version Requirement

KUDO version 0.4.0 or later

File: repository/kafka/docs/v0.2/install.md
Comment (zmalik) [reply repository/kafka/docs/v0.2/install.md:21]: I added an explanation around `ZOOKEEPER_PATH` its not the zookeeper connection uri but zk node that hosts the data.

File: repository/kafka/docs/latest/configuration.md
Comment (zmalik) [inline repository/kafka/docs/latest/configuration.md]: make it clear that zookeeper is a requirement. And connectivity to that zookeeper

File: repository/kafka/docs/latest/configuration.md
Comment (zmalik) [inline repository/kafka/docs/latest/configuration.md:74]: default zk path should include `namespace` value as a prefix for example

File: repository/kafka/docs/v0.2/limitations.md
Comment (zmalik) [inline repository/kafka/docs/v0.2/limitations.md:13]: These storage-related parameters cannot be changed after initial deployment.  using parameters to resize disk, change storage class, or switch between persistent/ephemeral storage is not supported.
```suggestion
These storage-related parameters cannot be changed after initial deployment. Using parameters to resize disk, change storage class, or switch between persistent/ephemeral storage is not supported.
```

File: repository/kafka/docs/v0.2/limitations.md
Comment (zmalik) [inline repository/kafka/docs/v0.2/limitations.md:3]: link the limitations from the installation section.

File: repository/kafka/docs/v0.2/limitations.md
Comment (zmalik) [inline repository/kafka/docs/v0.2/limitations.md]: Add a heading of Immutable parameters. and list the parameters in a list.

File: repository/kafka/docs/v0.2/limitations.md
Comment (zmalik) [inline repository/kafka/docs/v0.2/limitations.md:21]: link to the resizing PVC documentation of Kubernetes

File: repository/kafka/docs/v0.2/monitoring.md
Comment (zmalik) [inline repository/kafka/docs/v0.2/monitoring.md]: link to `service-monitor` resource from here

File: repository/kafka/docs/v0.2/monitoring.md
Comment (zmalik) [inline repository/kafka/docs/v0.2/monitoring.md:9]: mention metrics config yaml here

File: repository/kafka/docs/v0.2/release-notes.md
Comment (zmalik) [inline repository/kafka/docs/v0.2/release-notes.md]: change the date

File: repository/kafka/docs/v0.2/monitoring.md
Comment (zmalik) [inline repository/kafka/docs/v0.2/monitoring.md]: add link to prometheus operator

File: repository/kafka/docs/v0.2/update.md
Comment (zmalik) [inline repository/kafka/docs/v0.2/update.md]: link to limitations

**Review by kensipe — APPROVED** [review]

**Comment by jbarrick-mesosphere** [comment]
Fixed up labels to match what Prow expects

**Review by zmalik — COMMENTED** [review]
going to do few changes here

File: repository/kafka/docs/latest/configuration.md
Comment (zmalik) [reply repository/kafka/docs/latest/configuration.md:74]: we will evaluate adding this later and not at this stage.

File: repository/kafka/README.md
Comment (alenkacz) [inline repository/kafka/README.md]: 0.2.0 ?

File: repository/kafka/docs/latest/configuration.md
Comment (alenkacz) [inline repository/kafka/docs/latest/configuration.md]: I personally like docs to be written in simple language so I would not use the word ensemble as I had to google it :)

File: repository/kafka/docs/latest/configuration.md
Comment (alenkacz) [inline repository/kafka/docs/latest/configuration.md]: or you can use any other zookeeper cluster you're running inside or outside kubernetes? 🤔

File: repository/kafka/docs/latest/configuration.md
Comment (alenkacz) [inline repository/kafka/docs/latest/configuration.md]: too much KUDO kafka in this sentence, I don't understand. Maybe the second usage of that words can be dropped?

File: repository/kafka/docs/latest/configuration.md
Comment (alenkacz) [inline repository/kafka/docs/latest/configuration.md]: available via following DNS names?

File: repository/kafka/docs/latest/configuration.md
Comment (alenkacz) [inline repository/kafka/docs/latest/configuration.md]: `Let's for example see what will the zk path` - are we missing some verb here maybe? did you mean `look like`?

File: repository/kafka/docs/latest/release-notes.md
Comment (alenkacz) [inline repository/kafka/docs/latest/release-notes.md]: to prevent updating storage parameters?

File: repository/kafka/docs/latest/versions.md
Comment (alenkacz) [inline repository/kafka/docs/latest/versions.md:10]: is there something in kafka that will not work with 0.3.3? I think it does not matter at this point so leave it as is, just curious...

**Review by alenkacz — APPROVED** [review]
Just couple of small nits, looks good in general 👍

File: repository/kafka/docs/latest/configuration.md
Comment (zmalik) [reply repository/kafka/docs/latest/configuration.md]: it comes from DC/OS docs

File: repository/kafka/docs/latest/configuration.md
Comment (zmalik) [reply repository/kafka/docs/latest/configuration.md]: right, you can use as long as there is connectivity

File: repository/kafka/docs/latest/configuration.md
Comment (zmalik) [reply repository/kafka/docs/latest/configuration.md]: I introduced this because initially we had some confusion that `ZOOKEEPER_URI` is something that configures KUDO Zookeeper

File: repository/kafka/docs/latest/configuration.md
Comment (zmalik) [reply repository/kafka/docs/latest/configuration.md]: using word `cluster` now

File: repository/kafka/docs/latest/versions.md
Comment (zmalik) [reply repository/kafka/docs/latest/versions.md:10]: we just want users to not to lock in `0.3.3` for example. So adding as a requirement here.

**Pull request merged**
